### ISSUE TITLE
Conversion of spans leads to invalid blocks and drops images

### ISSUE THREAD

**Issue by pbauer** [plone/blocks-conversion-tool#21]
**Source is an empty span:**
```
<p><span></span></p>
```

Result:
```
{'@type': 'slate',
 'plaintext': '',
 'value': [{'children': [{'children': [], 'type': 'span'}], 'type': 'p'}]}
```
Problem: 
On editing it raises `Error: Cannot get the start point in the node at path [0] because it has no start text node.`

**Source is a span with a image** 
```
<p><span><img src='image.jpeg'></span></p>
```

Result (same as above):
```
{'@type': 'slate',
 'plaintext': '',
 'value': [{'children': [{'children': [], 'type': 'span'}], 'type': 'p'}]}
```
Problem: 
The image is dropped. On editing it again raises `Error: Cannot get the start point in the node at path [0] because it has no start text node.`

**Source is a span with image and text**
```
<p><span><img src='image.jpeg'>text</span></p>
```

Result
```
{'@type': 'slate',
 'plaintext': 'text',
 'value': [{'children': [{'children': [{'text': 'text'}], 'type': 'span'}], 'type': 'p'}]}
```

Problem: 
The image is dropped, the block is valid

**Comment by davisagli** [comment]
Similar to #20 -- thanks for the specific case

**Comment by pbauer** [comment]
The new results don't seem right. The images are still dropped.
 
All these do not return a image-block:
`<div><img src="image.jpeg" /></div>` 
`<span><img src="image.jpeg" /></span>` 
`<img src="image.jpeg" />` 
They are only created when they exits within a `<p>`

The changes span seems to work but I expected it to be dropped.
So instead `{'type': 'p', 'children': [{'text': ''}]}` is now is `{'type': 'p', 'children': [{'type': 'span', 'children': [{'text': ''}]}]}`

**Comment by ericof** [comment]
Looking into it

**Comment by ericof** [comment]
@pbauer Additional tests implemented on Release 0.4.6 (And on main)
